tests/Doctrine/Tests/Persistence/Mapping/ClassMetadataFactoryTest.php

```diff
@@ -141,20 +168,84 @@ public function testWillFailOnFallbackFailureWithNotLoadedMetadata(): void
      */
     public function testWillIgnoreCacheEntriesThatAreNotMetadataInstances(): void
     {
-        $cacheDriver = $this->createMock(Cache::class);
-        assert($cacheDriver instanceof Cache || $cacheDriver instanceof MockObject);
-
-        $this->cmf->setCacheDriver($cacheDriver);
-
-        $cacheDriver->expects(self::once())->method('fetch')->with('Foo$CLASSMETADATA')->willReturn(new stdClass());
+        $key = $this->cmf->getCacheKey(RootEntity::class);
+
+        $metadata = $this->cmf->metadata;
+        assert($metadata instanceof ClassMetadata || $metadata instanceof MockObject);
+
+        $item = $this->createMock(CacheItemInterface::class);
+        assert($item instanceof CacheItemInterface || $item instanceof MockObject);
+
+        $item
+            ->expects(self::any())
+            ->method('get')
+            ->willReturn(new stdClass());
+        $item
+            ->expects(self::once())
+            ->method('set')
+            ->with($metadata);
+
+        $cacheDriver = $this->createMock(CacheItemPoolInterface::class);
+        assert($cacheDriver instanceof CacheItemPoolInterface || $cacheDriver instanceof MockObject);
+        $cacheDriver
+            ->expects(self::any())
+            ->method('getItem')
+            ->with($key)
+            ->willReturn($item);
+        $cacheDriver
+            ->expects(self::once())
+            ->method('saveDeferred')
+            ->with($item);
+        $cacheDriver
+            ->expects(self::once())
+            ->method('commit');
+
+        $this->cmf->setCache($cacheDriver);
+
+        self::assertSame($metadata, $this->cmf->getMetadataFor(RootEntity::class));
+    }
 
-        $metadata = $this->createMock(ClassMetadata::class);
-        assert($metadata instanceof ClassMetadata);
+    public function testWillNotCacheFallbackMetadata(): void
```

Comment:
I duplicated that test, as I realised that using the `onMetadataNotFound` fallback does not cache that metadata. Since I assume this is intended, I haven't made changes and instead opted to codify that behaviour in tests.